lib/logger/utils.ts

```diff
@@ -90,38 +91,54 @@ export default function prepareError(err: Error): Record<string, unknown> {
   return response;
 }
 
-export function sanitizeValue(_value: unknown, seen = new WeakMap()): any {
-  let value = _value;
-  if (Array.isArray(value)) {
-    const length = value.length;
-    const arrayResult = Array(length);
-    seen.set(value, arrayResult);
-    for (let idx = 0; idx < length; idx += 1) {
-      const val = value[idx];
-      arrayResult[idx] = seen.has(val)
-        ? seen.get(val)
-        : sanitizeValue(val, seen);
-    }
-    return arrayResult;
+type NestedValue = unknown[] | object;
+
+function isNested(value: unknown): value is NestedValue {
+  return is.array(value) || is.object(value);
+}
+
+export function sanitizeValue(
+  value: unknown,
+  seen = new WeakMap<NestedValue, unknown>()
+): any {
+  if (is.string(value)) {
+    return sanitize(value);
   }
 
-  if (value instanceof Buffer) {
-    return '[content]';
+  if (is.date(value)) {
+    return value;
   }
 
-  if (value instanceof Error) {
-    value = prepareError(value);
+  if (is.function_(value)) {
+    return '[function]';
   }
 
-  const valueType = typeof value;
+  if (is.buffer(value)) {
+    return '[content]';
+  }
+
+  if (is.error(value)) {
+    const err = prepareError(value);
+    return sanitizeValue(err, seen);
+  }
 
-  if (value && valueType !== 'function' && valueType === 'object') {
-    if (value instanceof Date) {
-      return value;
+  if (is.array(value)) {
+    const length = value.length;
+    const arrayResult = Array(length);
+    seen.set(value, arrayResult);
+    for (let idx = 0; idx < length; idx += 1) {
+      const val = value[idx];
+      arrayResult[idx] =
+        isNested(val) && seen.has(val)
+          ? seen.get(val)
+          : sanitizeValue(val, seen);
     }
+    return arrayResult;
+  }
 
+  if (is.object(value)) {
```

Comment:
This branch can be also improved with further PR, I think particular field checks should be refactored out